includes/class-wc-google-gtag-js.php

```diff
@@ -28,6 +31,33 @@ public static function get_instance( $options = array() ) {
 	 */
 	public function __construct( $options = array() ) {
 		self::$options = $options;
+		// Setup frontend scripts
+		add_action( 'wp_enqueue_scripts', array( $this, 'register_scripts' ) );
+		add_action( 'woocommerce_before_single_product', array( $this, 'setup_frontend_scripts' ) );
+	}
+
+	/**
+	 * Enqueue the frontend scripts and make formatted variant data available via filter
+	 *
+	 * @return void
+	 */
+	public function setup_frontend_scripts() {
+		global $product;
+
+		if ( $product->is_type( 'variable' ) ) {
+			// Filter variation data to include formatted strings required for add_to_cart event
+			add_filter( 'woocommerce_available_variation', array( $this, 'variant_data' ), 10, 3 );
+			// Add default inline product data for add to cart tracking
+			wp_enqueue_script( $this->script_handle );
+			wp_add_inline_script( $this->script_handle, $this->default_add_to_cart_data( $product ) );
+		}
+	}
+
+	/**
+	 * Register front end JavaScript
+	 */
+	public function register_scripts() {
+		wp_register_script( $this->script_handle, plugins_url( 'assets/js/ga-integration.js', dirname( __FILE__ ) ), array( 'jquery' ), WC_GOOGLE_ANALYTICS_INTEGRATION_VERSION, true );
```

Comment:
I wasn't aware of the differences in `plugin_basename` so TIL and thanks for pointing that out, however, using `plugins_url` here so `dirname( __DIR__ )` would result in a broken path for the script so it would need to use `__DIR__` or `dirname( __FILE__ )`

Reply:
Ok, I see what's happening there, `plugins_url` is asking for any file within the root plugin directory. It strips off the file to get the directory name. So `includes` is seen as the file which is stripped. A bit confusing in my opinion but it works so we can leave it as is. In other plugins the main plugin file is passed to a class through a constant: https://github.com/woocommerce/automatewoo/blob/5.6.0/automatewoo.php#L41